percona/controller/pgcluster/controller.go

```diff
@@ -282,6 +286,99 @@ func (r *PGClusterReconciler) Reconcile(ctx context.Context, request reconcile.R
 	return ctrl.Result{}, nil
 }
 
+func (r *PGClusterReconciler) reconcileTLS(ctx context.Context, cr *v2.PerconaPGCluster) error {
+	checkSecretProjection := func(p *corev1.SecretProjection, requiredPaths ...string) error {
+		if p == nil {
+			return nil
+		}
+
+		if p.Name == "" {
+			return errors.New("secret name is not specified")
+		}
+
+		secret := new(corev1.Secret)
+		nn := types.NamespacedName{Name: p.Name, Namespace: cr.Namespace}
+		if err := r.Client.Get(ctx, nn, secret); err != nil {
+			return errors.Wrapf(err, "failed to get secret %s", nn.Name)
+		}
+
+		pathMap := make(map[string]struct{})
+		for _, item := range p.Items {
+			if _, ok := secret.Data[item.Key]; !ok {
+				return errors.Errorf("key %s doesn't exist in secret %s", item.Key, secret.Name)
+			}
+			pathMap[item.Path] = struct{}{}
+		}
+
+		for _, path := range requiredPaths {
+			if _, ok := pathMap[path]; !ok {
+				return errors.Errorf("required path %s was not found", path)
+			}
+		}
+
+		return nil
+	}
+
+	if err := checkSecretProjection(cr.Spec.Secrets.CustomRootCATLSSecret, "root.crt", "root.key"); err != nil {
+		return errors.Wrap(err, "failed to validate .spec.customRootCATLSSecret")
+	}
+
+	certPaths := []string{"tls.key", "tls.crt"}
+	if cr.Spec.Secrets.CustomRootCATLSSecret == nil {
+		certPaths = append(certPaths, "ca.crt")
+	}
+	if err := checkSecretProjection(cr.Spec.Secrets.CustomTLSSecret, certPaths...); err != nil {
+		return errors.Wrap(err, "failed to validate .spec.customTLSSecret")
+	}
+	if err := checkSecretProjection(cr.Spec.Secrets.CustomReplicationClientTLSSecret, certPaths...); err != nil {
+		return errors.Wrap(err, "failed to validate .spec.customReplicationTLSSecret")
+	}
+
+	oldCASecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      naming.RootCertSecret,
+			Namespace: cr.Namespace,
+		},
+	}
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(oldCASecret), oldCASecret)
+	if client.IgnoreNotFound(err) != nil {
+		return errors.Wrap(err, "failed to get old ca secret")
+	}
+
+	if cr.CompareVersion("2.5.0") < 0 {
+		if k8serrors.IsNotFound(err) {
+			// K8SPG-555: We should create an empty secret with old name, so that crunchy part can populate it
+			// instead of creating secrets unique to the cluster
+			// TODO: remove when 2.4.0 will become unsupported
+			if err := r.Client.Create(ctx, oldCASecret); err != nil {
+				return errors.Wrap(err, "failed to create ca secret")
+			}
+		}
+		return nil
+	}
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
```

Comment:
can we merge these two conditions:

```suggestion
	if k8serrors.IsNotFound(err) {
		if cr.CompareVersion("2.5.0") < 0 {
			// K8SPG-555: We should create an empty secret with old name, so that crunchy part can populate it
			// instead of creating secrets unique to the cluster
			// TODO: remove when 2.4.0 will become unsupported
			if err := r.Client.Create(ctx, oldCASecret); err != nil {
				return errors.Wrap(err, "failed to create ca secret")
			}
		}
		return nil
	}
```